### PR TITLE
upgrade to beta.3

### PR DESCRIPTION
**Pull request by sumitarora**

### CODE REVIEW

File: backend/adapters/base.ts [hunk 1]
```diff
@@ -42,10 +42,6 @@ export interface TreeNode {
   state: Object;
   input: Object;
   output: Object;
```
Comment (igor-ka) [inline backend/adapters/base.ts]: Were we populating lastTickTime before? If so, maybe it's worth keeping it?

Comment (sumitarora) [inline backend/adapters/base.ts]: nope it always returned 0 
```js
  _getComponentPerf(compEl: DebugElement): number {
    return 0;
  }
```

Comment (igor-ka) [inline backend/adapters/base.ts]: :+1:

**Comment by igor-ka** [comment]
@sumitarora I got this branch and went through the kitchen sink example with this code. Everything works either better or as good as before. 

In terms of code I'll leave a few comments. Its up to you if you want to address any of those in a separate PR. 

I will be using this for the webinar tomorrow.

File: backend/adapters/angular2.ts [hunk 2]
```diff
@@ -29,26 +29,64 @@ declare var ng: { probe: Function };
...
+
   setup(): void {
     // only supports applications with single root for now
     const root = this._findRoot();
-    this._traverseTree(ng.probe(root),
-                       this._emitNativeElement,
-                       true,
-                       '0');
+
+    // this._traverseTree(ng.probe(root),
```
Comment (igor-ka) [inline backend/adapters/angular2.ts]: I think we should remove the commented out code.

File: backend/adapters/angular2.ts [hunk 3]
```diff
@@ -78,24 +111,36 @@ export class Angular2Adapter extends BaseAdapter {
   }
 
   _findRoot(): Element {
-    return document.body.querySelector('[data-ngid]');
+    let elements: any = document.querySelectorAll('body *');
```
Comment (igor-ka) [inline backend/adapters/angular2.ts]: should probably use const instead of let for `elements` and `debugElement`

**Comment by sumitarora** [comment]
@igor-ka Yes I would do it in different PR with some refactoring and upgrades too :smile:

File: backend/adapters/angular2.ts [hunk 4]
```diff
@@ -241,37 +292,35 @@ export class Angular2Adapter extends BaseAdapter {
...
-    //     if (provider instanceof DirectiveProvider) {
-    //       props[provider.displayName] = provider.metadata.events;
-    //     }
-    //   }
-    // }
+  _getComponentInput(compEl: any): Object {
+    let props = [];
+    if (compEl.providerTokens.length > 0) {
+      try {
+        const directiveResolver: DirectiveResolver = new DirectiveResolver();
```
Comment (igor-ka) [inline backend/adapters/angular2.ts]: dont need `: DirectiveResolver`, as the type will be inferred by the compiler

File: example-apps/kitchen-sink-example/package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
   },
   "dependencies": {},
   "devDependencies": {
-    "angular2": "2.0.0-beta.1",
+    "angular2": "2.0.0-beta.3",
```
Comment (igor-ka) [inline example-apps/kitchen-sink-example/package.json]: zone.js dependency needs to be upgraded to 0.5.11 as there is a warning when building and running it that is related

File: example-apps/kitchen-sink-example/source/app.ts [hunk 6]
```diff
@@ -9,19 +9,25 @@ import {bootstrap} from 'angular2/bootstrap';
 import KitchenSink from './containers/kitchen-sink';
 
 // For Batarangle Integration
-import {AppViewListener} from 'angular2/src/core/linker/view_listener';
-import {DebugElementViewListener, inspectNativeElement}
-  from 'angular2/platform/common_dom';
+// import {AppViewListener} from 'angular2/src/core/linker/view_listener';
```
Comment (igor-ka) [inline example-apps/kitchen-sink-example/source/app.ts]: This is great, no need to modify your application to run it with Batarangle.
:+1:

Comment (igor-ka) [inline example-apps/kitchen-sink-example/source/app.ts]: But we should remove the commented out code

File: package.json [hunk 7]
```diff
@@ -33,7 +33,7 @@
     "pack": "./crxmake.sh"
   },
   "dependencies": {
-    "angular2": "^2.0.0-beta.2",
+    "angular2": "^2.0.0-beta.3",
```
Comment (igor-ka) [inline package.json]: same zone.js dependency that needs upgrading

**Comment by igor-ka** [comment]
@sumitarora as per my previous comment, LGTM to :shipit: if you want to address the comments in a separate PR.

**Pull request merged**